cost-analyzer/templates/cost-analyzer-deployment-template.yaml

```diff
@@ -643,7 +654,11 @@ spec:
             {{- if .Values.thanos.query }}
             {{- if .Values.thanos.query.enabled }}
             - name: THANOS_ENABLED
+            {{- if .Values.hosted }}
+              value: "false"
+            {{- else }}
```

Comment:
Is it safe to say that `.hosted=true` and `.thanos.query.enabled=true` is a bad config case?

If so, it would be cool if we could somehow enforce this. Some brief Googling found mention of a template function called `fail` [1] but it appears to require some sort of additional library, so I'm not sure if its worth the effort?

[1] https://austindewey.com/2018/12/28/helm-tricks-input-validation-with-required-and-fail/

Reply:
@michaelmdresser Checkout this PR: https://github.com/kubecost/cost-analyzer-helm-chart/pull/1568

I used `fail` to ensure that a specific combination of flags were set. It works great.

Reply:
Superb. That would be great in this PR if my guess about config state is right.